ArmRobot/Assets/Scripts/Robot/RobotVisionCapture.cs

```diff
@@ -0,0 +1,111 @@
+
+        // we create game objects 
+        int maxNumberObjectsPerRow = (int)(1/(minimumDistanceWithoutObjects + scaleObject)); 
+        for (var i = 0; i < randomNumberGameObjects; i++) {
+            GameObject sphere  = GameObject.CreatePrimitive(PrimitiveType.Sphere);
+
+            // we separate the case where i < maxNumberObjectsPerRow to the case i > maxNumberObjectsPerRow 
+            // because no more than maxNumberObjectsPerRow objects can have the same z coordinate 
+            if (i < maxNumberObjectsPerRow){
+                sphere.transform.position = new Vector3(robotMaxReach - minimumDistanceWithoutObjects - minimumDistanceBetweenObjects*i, 
+                                                    yAltitudeTable, robotMaxReach - minimumDistanceWithoutObjects);
+            }
+            else{
+                sphere.transform.position = new Vector3(robotMaxReach - minimumDistanceWithoutObjects - minimumDistanceBetweenObjects*(i-maxNumberObjectsPerRow+1), 
+                                                    yAltitudeTable, robotMaxReach - minimumDistanceWithoutObjects - minimumDistanceBetweenObjects);
+            }
+            
+            sphere.transform.localScale = new Vector3(scaleObject, scaleObject, scaleObject);
+            sphere.AddComponent<ColorRandomizer>();
+            sphere.AddComponent<RandomizerPositionObject>();
+            sphere.tag = "Sphere";
+            sphere.name = "Sphere_" + i;
+ }
+    }
+
+
+    private void Update()
+    {
+        VisionDataCollector visionDataCollector = GetComponent<VisionDataCollector>();
+        string imageName = visionDataCollector.NextImageName();
+        Vector3 relativeCubePosition = cube.transform.position - robot.transform.position;
+        RobotVisionDataPoint dataPoint = new RobotVisionDataPoint(relativeCubePosition, imageName);
+
+        bool didCapture = visionDataCollector.CaptureIfNecessary(imageName, dataPoint);
+
+        if (didCapture)
+        {
+            Reset();
+        }
+    }
+
+    // HELPERS
+
+    private void Reset()
+    {
+        // we create a list that will contains all the objects we have already moved 
+        List<GameObject> listOfAlreadyMovedObjects = new List<GameObject>();
+
+        // move cube
+        TablePositionRandomizer tablePositionRandomizer_cube = cube.GetComponent<TablePositionRandomizer>();
+        tablePositionRandomizer_cube.Move();
+
+        listOfAlreadyMovedObjects.Add(cube);
+
+        // move other objects 
+        // first we create an array of all the objects in the scene except the cube 
+
+        GameObject[] arrayOfSpheres= GameObject.FindGameObjectsWithTag("Sphere");
+        List<GameObject> listOfObjects = new List<GameObject>();
+        listOfObjects.AddRange(arrayOfSpheres); // we need to do that because we want to work with list and not array and arrayOfSpheres is an array 
+        listOfObjects.Add(cylinder);
```

Comment:
Are the number of spheres getting randomized? I don't see where that is happening.